File: src/main/java/com/cuupa/classificator/services/kb/semantic/token/OneOf.java
package com.cuupa.classificator.services.kb.semantic.token;

import com.cuupa.classificator.services.kb.semantic.text.TextSearch;
import org.jetbrains.annotations.NotNull;

import java.util.ArrayList;

public class OneOf extends Token {

	private int distance;

	@Override
	public boolean match(String text) {
        TextSearch textSearch = new TextSearch(text);
		for (String value : tokenValue) {
            if (textSearch.contains(value)) {
                distance = textSearch.getDistance();
				return true;
			}
		}
		return false;
	}

	public int getDistance() {
		return distance;
	}

	@NotNull
	@Override
	protected Token clone() {
		Token token = new OneOf();
		token.setTokenValue(new ArrayList<>(tokenValue));
		return token;
	}

	@NotNull
	@Override
	public String toString() {
		return "OneOf " + super.toString();
	}
}

File: src/main/java/com/cuupa/classificator/services/kb/semantic/token/OneOf.kt
package com.cuupa.classificator.services.kb.semantic.token

import com.cuupa.classificator.services.kb.semantic.text.TextSearch
import java.util.*

class OneOf : Token() {

    override var distance = 0
        private set

    override fun match(text: String?): Boolean {
        val textSearch = TextSearch(text)
        for (value in tokenValue) {
            if (textSearch.contains(value)) {
                distance = textSearch.distance
                return true
            }
        }
        return false
    }

    override fun clone(): Token {
        val token: Token = OneOf()
        token.tokenValue = ArrayList(tokenValue)
        return token
    }

    override fun toString(): String {
        return "OneOf " + super.toString()
    }

    companion object {
        const val name = "oneOf"
    }
}